src/class-parsely.php

```diff
@@ -608,4 +566,65 @@ public function get_all_track_types(): array {
 	public function get_default_options() {
 		return $this->option_defaults;
 	}
+
+	/**
+	 * Returns the credentials that are being managed at the platform level.
+	 *
+	 * @since 3.9.0
+	 * @access private
+	 *
+	 * @return Parsely_Options|array<empty> The managed credentials.
+	 */
+	private function get_managed_credentials() {
+		if ( true !== $this->are_credentials_managed ) {
+			return array();
+		}
+
+		$credentials = apply_filters( 'wp_parsely_credentials', false );
+		$result      = array();
+
+		if ( isset( $credentials['site_id'] ) ) {
+			$result['apikey'] = $credentials['site_id'];
+		}
+
+		if ( isset( $credentials['api_secret'] ) ) {
+			$result['api_secret'] = $credentials['api_secret'];
+		}
+
+		if ( isset( $credentials['metadata_secret'] ) ) {
+			$result['metadata_secret'] = $credentials['metadata_secret'];
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Returns whether credentials are being managed at the platform level.
+	 *
+	 * @since 3.9.0
+	 * @access private
+	 *
+	 * @return bool Whether credentials are being managed at the platform level.
+	 */
+	private function are_credentials_managed(): bool {
```

Comment:
> rename this method to check_for_managed_credentials() (or similar)

Alternate idea would be to call this `load_managed_credentials` and combine it with `get_managed_credentials`.

Specifically, after validating that the credentials are passed through, it can set `are_credentials_managed = true` and also a new property called `credentials`.

Main benefit is that we simplify things into 1-step and avoid having to maintain the filter in two places.

Again, not a blocker. Can be done later if we want to.